### PR TITLE
Add omnibox.onDeleteSuggestion

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Adds documentation for `omnibox.onDeleteSuggestion` along with some minor edits to the `omnibox` page. (BCD previously updated in  Add deletable, onDeleteSuggestion, improve accuracy of version_added [#18372](https://github.com/mdn/browser-compat-data/pull/18372)) 

### Related issues and pull requests

Addresses the documentation requirements of [Bug 1478095](https://bugzilla.mozilla.org/show_bug.cgi?id=1478095) Implement `onDeleteSuggestion`

### CODE REVIEW

**Comment by rebloor** [comment]
@rpl are you able to review this?

File: files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - : Check whether `listener` is registered for this event. Returns `true` if it is listening, `false` otherwise.
+
+## addListener syntax
+
+The listener function passes one parameter: a string `text`.
+
+### Parameters
+
+- `text`
+  - : `String`. The description of the deleted suggestion.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md:44]:
```suggestion
  - : `string`. The description of the deleted suggestion.
```

File: files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- `addListener(listener)`
+  - : Adds a listener to this event.
+- `removeListener(listener)`
+  - : Stop listening to this event. The `listener` argument is the listener to remove.
+- `hasListener(listener)`
+  - : Check whether `listener` is registered for this event. Returns `true` if it is listening, `false` otherwise.
+
+## addListener syntax
+
+The listener function passes one parameter: a string `text`.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md:39]:
```suggestion
```

File: files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- `text`
+  - : `String`. The description of the deleted suggestion.
+
+## Examples
+
+This example prints the deleted suggestion to the console:
+
+```js
+function logDeletedSuggestion(deletedSuggestionInfo) {
+  console.log(`The user deleted: ${deletedSuggestionInfo.text}`);
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md:52]:
```suggestion
function logDeletedSuggestion(text) {
  console.log(`The user deleted: ${text}`);
```

File: files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - Reference
+  - WebExtensions
+  - omnibox
+  - onDeleteSuggestion
+browser-compat: webextensions.api.omnibox.onDeleteSuggestion
+---
+
+{{AddonSidebar()}}
+
+Fired whenever the user deletes a suggestion.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/omnibox/ondeletesuggestion/index.md:18]:
```suggestion
Fired whenever the user deletes a suggestion.
A suggestion can be deleted when {{WebExtAPIRef("omnibox.SuggestResult","SuggestResult")}}`.deletable` is set to true.
```

**Review by Rob--W — CHANGES_REQUESTED** [review]
TODO:

- [ ] Document `deletable` property at https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/omnibox/SuggestResult (Chrome docs: https://developer.chrome.com/docs/extensions/reference/omnibox/#property-SuggestResult-deletable, Firefox source: https://searchfox.org/mozilla-central/rev/9de332d5c8faac58dc1232b8a6383ce6cb1400f4/browser/components/extensions/schemas/omnibox.json#58-62)
- [x] apply suggested edits from this review
- [ ]

**Review by Rob--W — APPROVED** [review]

**Pull request merged**